src/Storages/StorageURL.cpp

```diff
@@ -151,7 +152,11 @@ IStorageURLBase::IStorageURLBase(
     storage_metadata.setConstraints(constraints_);
     storage_metadata.setComment(comment);
     setInMemoryMetadata(storage_metadata);
-    setVirtuals(VirtualColumnUtils::getVirtualsForFileLikeStorage(storage_metadata.getColumns()));
+
+    std::string uri_for_partitioning;
+    if (context_->getSettingsRef().use_hive_partitioning)
+        uri_for_partitioning = uri;
```

Comment:
`uri` can contain hlobs here like `url('http://localhost:11111/test/hive_partitioning/column0={1...100}/sample.parquet')`
And we will extract key `{1...100}` and won't infer its type right. 
We should do:

```c++
if (urlWithGlobs(uri))
{
    auto uris = parseRemoteDescription(uri, 0, uri.size(), ',', context->getSettingsRef().glob_expansion_max_elements);
    if (!uris.empty())
        uri_for_partitioning = uri;
}
else
{
    uri_for_partitioning = uri;
}
```

And maybe move it to a separate function `getSampleURI`.